qualang_tools/voltage_gates/voltage_gate_sequence.py

```diff
@@ -230,27 +231,83 @@ def add_step(
                             wait(_duration >> 2, gate)
             self.current_level[i] = voltage_level
 
-    def add_compensation_pulse(self, duration: int) -> None:
-        """Add a compensation pulse of the specified duration whose amplitude is derived from the previous operations.
-
-        :param duration: Duration of the compensation pulse in clock cycles (4ns). Must be larger than 4 clock cycles.
+    def add_compensation_pulse(self, max_amplitude: float = 0.49, **kwargs) -> None:
+        """Add a compensation pulse of the specified amplitude whose duration is derived automatically from the previous operations and the maximum amplitude allowed.
+        Note that the derivation of the compensation pulse parameters in real-time may add a gap up to 300ns before playing the pulse, but the voltage will be maintained.
+        :param max_amplitude: Maximum amplitude allowed for the compensation pulse in V. Default is 0.35V.
         """
-        self._check_duration(duration)
+        duration = kwargs.get("duration", None)
+        if duration is not None:
+            warn(
+                "The duration argument is deprecated, from qualang-tools 0.20 the compensation pulse duration is derived automatically based on the maximum amplitude allowed.",
```

Comment:
Maybe "The duration argument is deprecated and will be ignored. From qualang-tools 0.20..."

Reply:
At the moment it is still being used if specified, but I wanted to indicate that it will be ignored soon so that people will stope using it.
Maybe something like "The duration argument is deprecated and will soon be ignored. From qualang-tools 0.20 the..." wdyt?

Reply:
It sounds good, thanks!